library/src/hcc_detail/hipblas.cpp

```diff
@@ -16627,6 +16628,122 @@ catch(...)
     return exception_to_hipblas_status();
 }
 
+hipblasStatus_t hipblasSgels(hipblasHandle_t    handle,
+                             hipblasOperation_t trans,
+                             const int          m,
+                             const int          n,
+                             const int          nrhs,
+                             float*             A,
+                             const int          lda,
+                             float*             B,
+                             const int          ldb,
+                             int*               info)
```

Comment:
I'd suggest making the method signature closer to what cuBLAS has for gelsBatched, which in this case is an additional info pointer on the host. It would be nice if you could set the value of this pointer similar to how it's done by hipBLAS's getrs and geqrf methods.

You might also want to consider only adding the batched method, like how we did for getri. Up to you.

Reply:
My understanding is that the two info pointers communicate different information. The first info is on the host and returns a value <= 0 depending on the presence of invalid arguments. The second info is on the device and returns a value >= 0 depending on singularities in the problem.

Reply:
Ah, my fault, I see the difference now, yes I'll update this and let you know.

Reply:
Added, haven't tested myself so will wait for CI. Let me know if the arg checking in hcc_detail/hipblas.cpp looks correct.